src/utils/sorting.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Compares two strings.
+ * Intended to be used as a comparator function in `Array.prototype.sort`.
```

Comment:
If I understand your intention correctly, you want to pass the `isAscending` param as a dynamic prop inside a closure, e.g. `.sort((a, b) => compareStrings(a, b, isAscendingState))`
If that's the usage, it's a bit problematic with memory performance as it re-allocates sort functions every time the `isAscendingState` changes for every single call.
What would be better is if you make them factories that return functions, i.e. `function compareStringsFactory(a, b, isAsc) { return (a, b) => isAsc? a.compare(b) : b.compare(a) }` then cache is with `useMemo` based on the state, to enable passing a direct reference: `.sort(compareStringsMemo)`